lua/telescope/utils.lua

```diff
@@ -82,6 +82,75 @@ utils.quickfix_items_to_entries = function(locations)
   return results
 end
 
+utils.filter_symbols = function(results, opts)
+  if opts.symbols == nil then
+    return results
+  end
+  local valid_symbols = vim.tbl_map(string.lower, vim.lsp.protocol.SymbolKind)
+
+  local filtered_symbols = {}
+  if type(opts.symbols) == "string" then
+    opts.symbols = string.lower(opts.symbols)
+    if vim.tbl_contains(valid_symbols, opts.symbols) then
+      for _, result in pairs(results) do
```

Comment:
Okay again why is this a `pairs` over a `ipairs`? Isn't `results` a list? `1 ... n` so `ipairs` should be better